### PR TITLE
fix(#48) local build of `@callstack/out-of-tree-platforms`

### PR DESCRIPTION
**Pull request by okwasniewski**
## Summary:

This PR should fix issues with running local RNTester app while packages are not built. For some reason RN core is not running a `postinstall` script that builds the packages so `@callstack/out-of-tree-platforms` wasn't built after cloning the repo. 

## Changelog:

[INTERNAL] [FIXED] - fix: local build of RNTester with `@callstack/out-of-tree-platforms`

## Test Plan:

CI Green

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
     "test-typescript": "dtslint packages/react-native/types",
     "test": "jest",
     "trigger-react-native-release": "node ./scripts/trigger-react-native-release.js",
-    "update-lock": "npx yarn-deduplicate"
+    "update-lock": "npx yarn-deduplicate",
+    "postinstall": "yarn build"
```
Comment (Marcoo09) [inline package.json:40]: Just curious, why is this necessary?

File: .prettierignore [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# Ignore packages dist
+packages/*/dist
```
Comment (okwasniewski) [inline .prettierignore:2]: This could be potentially upstreamed to Core

Comment (okwasniewski) [reply .prettierignore:2]: https://github.com/facebook/react-native/pull/41826

Comment (matthargett) [reply .prettierignore:2]: Upstream PR merged, does this PR close now?
